### PR TITLE
fix missing node modules during rsync copy

### PR DESCRIPTION
**Pull request by arroyc**
when we do rsync copy we exclude node_modules folder, but somehow we were excluding all the node_modules folder in all sub directories, rather we should exclude the node_modules folder only on the root.

### CODE REVIEW

File: tests/SampleApps/nodejs/kudu-bug/another-directory/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "kudu-bug",
```
Comment (kichalla) [inline tests/SampleApps/nodejs/kudu-bug/another-directory/package.json]: please rename this app to be more appropriate for the scenario being tested here...something which indicates `nested package.json`

File: tests/Oryx.Integration.Tests/Node/NodeTestUsingZippedNodeModules.cs [hunk 2]
```diff
@@ -164,5 +164,48 @@ public async Task BuildsAndRunsNodeApp_WhenPruneDevDependenciesIsTrue_AndNodeMod
                     Assert.Contains("Say It Again", data);
                 });
         }
+
+        [Fact]
+        public async Task CopiesNodeModulesInSubDirectory_ToDestination_WhenNodeModulesAreCompressed()
```
Comment (kichalla) [inline tests/Oryx.Integration.Tests/Node/NodeTestUsingZippedNodeModules.cs]: Should we even test when compression is not used?

File: tests/Oryx.Integration.Tests/Node/NodeTestUsingZippedNodeModules.cs [hunk 3]
```diff
@@ -164,5 +164,48 @@ public async Task BuildsAndRunsNodeApp_WhenPruneDevDependenciesIsTrue_AndNodeMod
...
+                .AddCommand($"oryx -appPath {appOutputDir} -bindPort {ContainerPort}")
+                .AddCommand(DefaultStartupFilePath)
+                .AddDirectoryExistsCheck($"{appOutputDir}/another-directory/node_modules")
+                .ToString();
+            var buildScript = new ShellScriptBuilder()
+               .AddCommand(
+                $"oryx build {appDir} -i /tmp/int -o {appOutputDir} --platform nodejs " +
+                $"--platform-version {nodeVersion} -p {NodePlatform.CompressNodeModulesPropertyKey}=tar-gz" +
+                $" -p {NodePlatform.PruneDevDependenciesPropertyKey}=true")
+               //.AddCommand($"cp -rf /tmp/out/* {appOutputDir}")
```
Comment (kichalla) [inline tests/Oryx.Integration.Tests/Node/NodeTestUsingZippedNodeModules.cs]: remove?

File: tests/Oryx.Integration.Tests/Node/NodeTestUsingZippedNodeModules.cs [hunk 4]
```diff
@@ -164,5 +164,48 @@ public async Task BuildsAndRunsNodeApp_WhenPruneDevDependenciesIsTrue_AndNodeMod
...
+            var runAppScript = new ShellScriptBuilder()
+                .AddCommand($"oryx -appPath {appOutputDir} -bindPort {ContainerPort}")
+                .AddCommand(DefaultStartupFilePath)
+                .AddDirectoryExistsCheck($"{appOutputDir}/another-directory/node_modules")
+                .ToString();
+            var buildScript = new ShellScriptBuilder()
+               .AddCommand(
+                $"oryx build {appDir} -i /tmp/int -o {appOutputDir} --platform nodejs " +
+                $"--platform-version {nodeVersion} -p {NodePlatform.CompressNodeModulesPropertyKey}=tar-gz" +
+                $" -p {NodePlatform.PruneDevDependenciesPropertyKey}=true")
```
Comment (kichalla) [inline tests/Oryx.Integration.Tests/Node/NodeTestUsingZippedNodeModules.cs]: Current app service does not use this, so i would say to to have multiple test cases here to test different scenarios here

File: src/BuildScriptGenerator/Node/NodePlatform.cs [hunk 5]
```diff
@@ -300,7 +300,7 @@ public IEnumerable<string> GetDirectoriesToExcludeFromCopyToIntermediateDir(Buil
                 NodeConstants.AllNodeModulesDirName,
                 NodeConstants.ProdNodeModulesDirName,
                 NodeConstants.NodeModulesDirName,
-                NodeConstants.NodeModulesToBeDeletedName,
+                "/"+NodeConstants.NodeModulesToBeDeletedName,
```
Comment (kichalla) [inline src/BuildScriptGenerator/Node/NodePlatform.cs]: Can you add a comment as to why we are doing this

File: src/BuildScriptGenerator/Node/NodeConstants.cs [hunk 6]
```diff
@@ -17,7 +17,7 @@ internal static class NodeConstants
         internal const string NpmStartCommand = "npm start";
         internal const string YarnStartCommand = "yarn run start";
         internal const string YarnCommand = "yarn";
-        internal const string NpmPackageInstallCommand = "npm install";
+        internal const string NpmPackageInstallCommand = "npm install --unsafe-perm";
```
Comment (kichalla) [inline src/BuildScriptGenerator/Node/NodeConstants.cs]: Does the test in this PR fail without this flag?

File: tests/Oryx.Integration.Tests/Node/NodeTestUsingZippedNodeModules.cs (hunk 2 above)
Comment (arroyc) [reply tests/Oryx.Integration.Tests/Node/NodeTestUsingZippedNodeModules.cs]: good idea

File: src/BuildScriptGenerator/Node/NodeConstants.cs (hunk 6 above)
Comment (arroyc) [reply src/BuildScriptGenerator/Node/NodeConstants.cs]: as discussed offline

Comment (kichalla) [inline src/BuildScriptGenerator/Node/NodeConstants.cs]: can you try and see if `yarn` also needs something like this?

**Pull request merged**